packages/react-router-dom/index.tsx

```diff
@@ -976,6 +977,95 @@ export function useFormAction(
   return createPath(path);
 }
 
+let blockerKey = "blocker-singleton";
+
+export function useBlocker(
+  shouldBlock: boolean | (() => boolean) | BlockerFunction
+) {
+  let { router } = useDataRouterContext(DataRouterHook.UseFetcher);
```

Comment:
Probably makes sense to add a new `DataRouterHook` here for UseBlocker if I understand the usage correctly.